File: exercism/javascript/complete/grade-school/grade-school.js
const School = () => {
  const kids = {};
  const roster = () => kids;
  const grade = year => kids[year] || [];
  const add = (name, year) => {
    kids[year] ? kids[year].push(name) : kids[year] = [name];
    kids[year].sort();
  };
  return { roster, grade, add };
};

export default School;

File: exercism/ruby/complete/grade-school/grade_school.rb
class School
  def initialize
    @school = []
    (1..12).each{ |num| @school << { grade: num, students: [] } }
  end

  def add(name, grade)
    @school[grade-1][:students] << name
    @school[grade-1][:students].sort!
  end

  def students(grade)
    @school[grade-1][:students]
  end

  def students_by_grade
    @school.select{ |grade| !grade[:students].empty? }
  end
end
